app/models/manager_refresh/inventory_collection.rb

```diff
@@ -186,8 +186,12 @@ def lazy_find(manager_uuid, key: nil, default: nil)
       ::ManagerRefresh::InventoryObjectLazy.new(self, manager_uuid, :key => key, :default => default)
     end
 
+    def inventory_object_class
+      @inventory_object_class ||= Class.new(::ManagerRefresh::InventoryObject)
```

Comment:
Yeah, so it's only way I found, where we will end up with the right readers/writers on a InventoryObject class, according to model. Otherwise, they were all adding readers/writers into the one InventoryObject class, so it ended up with methods of all models.

Is there some concern wit Class.new? It should be garbage collected as any other object, right?